### PR TITLE
Disable and deprecate `force-local-policy-eval-at-source`

### LINKED ISSUE #15452: Remove force-local-policy-eval-at-source flag
As its name indicates, the `force-local-policy-eval-at-source` flag force policy evaluation at the source (whether its bpf_lxc, bpf_host, or bpf_overlay): https://github.com/cilium/cilium/blob/a05461a3199b3da8d2558add032c8fffc7ff5c3d/bpf/lib/l3.h#L91-L105
Regardless of whether a BPF program is loaded on the egress of the lxc device (`USE_BPF_PROG_FOR_INGRESS_POLICY`), policies are enforced via a tail call. This causes packets to be processed twice when per-endpoint routes are enabled (cf. https://github.com/cilium/cilium/issues/14657).
The flag is currently enabled by default.

This flag was introduced in https://github.com/cilium/cilium/commit/c525c7555fc6ea84660a0a3721038f3b432caea8 and meant to introduce L7 issues on some kernels. Before disabling and removing that flag, we should:
- [x] Add policy tests to cilium connectivity test including L7 proxy tests.
- [x] Validate L7 proxy functionality with this flag disabled on various platforms (EKS, AKS, GKE).
- [x] Disable the flag by default.
- [x] Deprecate it.
- [ ] Remove it after another release in master.

### PR DESCRIPTION
**Pull request by pchaigno**
There isn't much more than what the title says here, but do check the commit descriptions of course.

Tested manually on GKE and AKS (given those workflows are disabled) with a 3-nodes cluster where I ran the connectivity tests 5 times (different pods each time). 

Fixes: https://github.com/cilium/cilium/issues/15452.

### CODE REVIEW

**Review by thorn3r — APPROVED** [review]
LGTM. some tests are failing, in particular GKE connectivity- but that seems to be from a cluster creation error (out of private IP space). should we kick that off again or good to go given your manual testing?

**Comment by pchaigno** [comment]
> some tests are failing, in particular GKE connectivity- but that seems to be from a cluster creation error (our of private IP space). should we kick that off again or good to go given your manual testing?

I'll triage the tests once the second review is in. From a quick look, they all appear to be known flakes.

File: daemon/cmd/daemon_main.go [hunk 1]
```diff
@@ -354,6 +354,7 @@ func initializeFlags() {
 
 	flags.Bool(option.ForceLocalPolicyEvalAtSource, defaults.ForceLocalPolicyEvalAtSource, "Force policy evaluation of all local communication at the source endpoint")
 	option.BindEnv(Vp, option.ForceLocalPolicyEvalAtSource)
+	flags.MarkDeprecated(option.ForceLocalPolicyEvalAtSource, fmt.Sprintf("This option will be removed in v1.14"))
```
Comment (AwesomePatrol) [inline daemon/cmd/daemon_main.go]: nit: `fmt.Sprintf` looks redundant here

Comment (pchaigno) [reply daemon/cmd/daemon_main.go]: Oops, right!

**Review by joestringer — APPROVED** [review]
Yep looks redundant vs. `USE_BPF_PROG_FOR_INGRESS_POLICY` (auto-enabled from the CNI in relevant environments). This combined with your testing is good enough to convince me :+1:

**Comment by julianwiedmann** [comment]
If I understand commit c525c7555fc6 correctly, one concern back then was the reconfig scenario where per-EP-routes gets switched on. And the source program (eg. `bpf_overlay`) has already been regenerated and trusts that that the pod will enforce policy in `to-container`... but the pod hasn't been regenerated yet, and so there's no `to-container` program installed yet.

That feels like a legit concern - but maybe it's no longer valid in today's code base?

**Comment by pchaigno** [comment]
> the reconfig scenario where per-EP-routes gets switched on

We've discussed this and concluded that switching the endpoint routes setting isn't supported today without downtime. We would need additional logic to support it.

**Comment by julianwiedmann** [comment]
> > the reconfig scenario where per-EP-routes gets switched on
> 
> We've discussed this and concluded that switching the endpoint routes setting isn't supported today without downtime. We would need additional logic to support it.

Makes sense :+1:. Would be nice to have this conclusion in the docs somewhere.

**Comment by pchaigno** [comment]
I've pushed [a small change](https://github.com/cilium/cilium/compare/5a7a34bfec7796c2382a291713f24e8865dd27ca..a59c368970cfa6f4291e4ba663c5d615c1a4d372) to address @AwesomePatrol's review and didn't rebase. As illustrated by MLH's label, all tests were passing before this change. Given the change is trivial and everything still compiles, I think it's fine to merge without rerunning the full CI.

**Pull request merged**